kurtosis-module/impl/participant_network/mev_boost/mev_boost_context.go

```diff
@@ -1,5 +1,20 @@
 package mev_boost
 
+import (
+	"fmt"
+
+	"github.com/kurtosis-tech/kurtosis-core-api-lib/api/golang/lib/services"
+)
+
 type MEVBoostContext struct {
-	// TODO Information about the mev-boost node that the CL client needs, analogous to ELClientContext
+	service *services.ServiceContext
+}
+
+func (ctx *MEVBoostContext) Endpoint() string {
+	ports := ctx.service.GetPrivatePorts()
+	port, ok := ports["api"]
+	if !ok {
+		panic("invariant violated, check port handling")
```

Comment:
We generally try to avoid panicking and instead handle the errors gracefully (which will give us stacktraces plus more debugging information). To accomplish this, I'd recommend making MEVBoost simpler, like so:

```golang
type MEVBoostContext struct {
    endpointUrl string
}

func NewMEVBoostContext(endpointUrl string) {
     ....instantiate
}

func (ctx *MEVBoostContext) GetEndpoint() {
    return ctx.endpointUrl
}
```

The reasons we'd do it this way, rather than putting the whole serviceCtx inside the struct:
1) the error-handling for the serviceCtx port-getting can be done outside this struct, in an area that returns an `error` (which allows us to use the `stacktrace` library to get nice stacktraces)
2) the constructor-getter pattern allows MEVBoostContext to be immutable from outside this package, so easier to debug
3) the port ID might not be `"api"` in perpetuity, so it allows the caller to pass in the right URL (rather than assuming the port ID is `"api"`)

Then you just need to push the URL-building and the

Reply:
I think dropping the serviceCtx is a good idea, adjusted in updates